x-pack/solutions/observability/plugins/observability/public/pages/alert_details/explain_alert_button.tsx

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiButton } from '@elastic/eui';
+import { ALERT_RULE_PARAMETERS } from '@kbn/rule-data-utils';
+import dedent from 'dedent';
+import type { AlertDetailsContextualInsight } from '../../../server/services';
+import type { AlertData } from '../../hooks/use_fetch_alert_detail';
+import { useKibana } from '../../utils/kibana_react';
+
+export function AlertDetailInvestigateButton({ alert }: { alert: AlertData | null }) {
+  const {
+    services: { onechat, http },
+  } = useKibana();
+
+  const [loading, setLoading] = useState(false);
+
+  const onClick = useCallback(async () => {
+    if (!alert) {
+      return;
+    }
+    try {
+      setLoading(true);
+      const fields = alert.formatted.fields as Record<string, unknown> | undefined;
+      const reason = alert.formatted.reason;
+
+      let obsAlertContext = '';
+      try {
+        const { alertContext } = await http.get<{ alertContext: AlertDetailsContextualInsight[] }>(
+          '/internal/observability/assistant/alert_details_contextual_insights',
+          {
+            query: {
+              alert_started_at: new Date(alert.formatted.start).toISOString(),
+              alert_rule_parameter_time_size: alert.formatted.fields[ALERT_RULE_PARAMETERS]
+                ?.timeSize as string | undefined,
+              alert_rule_parameter_time_unit: alert.formatted.fields[ALERT_RULE_PARAMETERS]
+                ?.timeUnit as string | undefined,
+              'service.name': (fields?.['service.name'] as string) || undefined,
+              'service.environment': (fields?.['service.environment'] as string) || undefined,
+              'transaction.type': (fields?.['transaction.type'] as string) || undefined,
+              'transaction.name': (fields?.['transaction.name'] as string) || undefined,
+              'host.name': (fields?.['host.name'] as string) || undefined,
+              'container.id': (fields?.['container.id'] as string) || undefined,
+              'kubernetes.pod.name': (fields?.['kubernetes.pod.name'] as string) || undefined,
+            },
```

Comment:
For the new API we should just pass the `alert.id` as query param. The API will then fetch the alert.